src/App.tsx

```diff
@@ -1,19 +1,64 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { FilterType } from './types/FilterType';
+// import { User } from './types/User';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[] | undefined>();
+  const [errorMessage, setErrorMessage] = useState('');
+  const [errorIsHidden, setErrorIsHidden] = useState(true);
+  const [filter, setFilter] = useState(FilterType.All);
+  const visibleTodos = useMemo<Todo[] | []>(() => {
```

Comment:
```suggestion
  const visibleTodos = useMemo<Todo[]>(() => {
```

An emptry array is also Todo[]